src/main/java/service/impl/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package service.impl;
+
+import db.FruitShopDao;
+import java.util.List;
+import model.FruitTransaction;
+import service.FruitTransactionService;
+import strategy.FruitStrategy;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private FruitShopDao fruitShopDao;
+    private FruitStrategy fruitStrategy;
+
+    public FruitTransactionServiceImpl(FruitShopDao fruitShopDao, FruitStrategy fruitStrategy) {
+        this.fruitShopDao = fruitShopDao;
+        this.fruitStrategy = fruitStrategy;
+    }
+
+    @Override
+    public String processTransactions(List<FruitTransaction> transactions) {
+        if (transactions == null) {
+            throw new RuntimeException("Incorrect data passed!");
+        }
+        for (FruitTransaction transaction : transactions) {
+            processSingleTransaction(transaction);
```

Comment:
Better move db change to the `handleOperation` method. 
```suggestion
            fruitStrategy.getOperationHandler(fruitTransaction.getOperation())
                .handleOperation(transaction);
```